### PR TITLE
UI: merge directly from import wizard

### LINKED ISSUE #5348: Smooth UI import screen workflow
On the Import UI, have we considered changing the Import completed button (which is permanently disabled?) to instead take the user to the merge screen so that they can review/commit the import?

image

Also, since `lakectl` has the `--merge` option to automagically do the merge, should this be a checkbox on the UI too? It could make the process smoother.

### PR DESCRIPTION
**Pull request by ozkatz**
### Linked Issue

Closes #5348 
      
### New Feature

Allow merging directly after the import modal completes importing.
Additionally, this PR also introduces a small enhancement in the form of commit metadata fields. Mostly for parity with the regular commit modal from the changes tab.

## Additional info

How it looks now:

screen

### Also

Removed dead code (old getting started wizard)

### CODE REVIEW

File: webui/src/lib/components/repository/changes.jsx [hunk 1]
```diff
@@ -280,3 +283,48 @@ const ExperimentalDeltaDiffButton = ({showButton = false}) => {
...
+export const MetadataFields = ({ metadataFields, setMetadataFields}) => {
+    return (
+        <div className="mt-3">
+            {metadataFields.map((f, i) => {
+                return (
+                    <Form.Group key={`commit-metadata-field-${f.key}-${f.value}-${i}`} className="mb-3">
+                        <Row>
+                            <Col md={{span: 5}}>
+                                <Form.Control type="text" placeholder="Key" defaultValue={f.key}
+                                              onChange={(e) => {
```
Comment (eladlachmi) [inline webui/src/lib/components/repository/changes.jsx:296]: You shouldn't mutate props like this. A good way of thinking about props is as read-only. Want to make a change? Create a new one with your changes and pass that in as the parameters to the setter function. If you need to do multiple levels of spread and it gets messy, you can use something like [Immer](https://immerjs.github.io/immer/), but I'm not sure that's needed at this point.  
While you're at it, maybe move that anonymous function to a named function with `useCallback`? 😉

File: webui/src/lib/components/repository/changes.jsx [hunk 2]
```diff
@@ -280,3 +283,48 @@ const ExperimentalDeltaDiffButton = ({showButton = false}) => {
...
+                                                  metadataFields[i].key = e.currentTarget.value;
+                                                  setMetadataFields(metadataFields);
+                                              }}/>
+                            </Col>
+                            <Col md={{span: 5}}>
+                                <Form.Control type="text" placeholder="Value" defaultValue={f.value}
+                                              onChange={(e) => {
+                                                  metadataFields[i].value = e.currentTarget.value;
```
Comment (eladlachmi) [inline webui/src/lib/components/repository/changes.jsx:304]: Same as above.

File: webui/src/pages/repositories/repository/changes.jsx [hunk 3]
```diff
@@ -67,44 +64,7 @@ const CommitButton = ({repo, onCommit, enabled = false}) => {
...
-                            )
-                        })}
-
-                        <Button onClick={() => {
-                            setMetadataFields([...metadataFields, {key: "", value: ""}]);
-                        }} size="sm" variant="secondary">
-                            <PlusIcon/>{' '}
-                            Add Metadata field
-                        </Button>
+                        <MetadataFields metadataFields={metadataFields} setMetadataFields={setMetadataFields}/>
```
Comment (eladlachmi) [inline webui/src/pages/repositories/repository/changes.jsx:67]: Like! 🤩
Cleanup FTW

File: webui/src/pages/repositories/services/import_data.jsx [hunk 4]
```diff
@@ -100,28 +103,63 @@ const ImportDone = ({numObjects, importBranch, currBranch = ''}) => {
...
+const ExecuteImportButton = ({isEnabled, importPhase, importFunc, mergeFunc, doneFunc}) => {
+    switch (importPhase) {
+        case ImportPhase.Completed:
+            return <Button
+                variant="success"
+                onClick={mergeFunc}
+                disabled={!isEnabled}>
+                    Merge Changes
+            </Button>
+        case ImportPhase.NotStarted:
```
Comment (eladlachmi) [inline webui/src/pages/repositories/services/import_data.jsx:122]: You can merge `NotStarted` and `Failed` into the same case if you want
Something like:
```JavaScript
case ImportPhase.NotStarted:
case ImportPhase.Failed:
  return <Button
                variant="success"
                disabled={!isEnabled}
                onClick={importFunc}>
                Import
            </Button>
```

**Review by eladlachmi — CHANGES_REQUESTED** [review]
Looks good overall.
Mostly minor comments. The most significant issue is mutating props.

File: webui/src/lib/components/repository/changes.jsx (hunk 1 above)
Comment (ozkatz) [reply webui/src/lib/components/repository/changes.jsx:296]: Honestly I just moved existing code and wrapped it in its own component - didn't really dig into how it works. I'll give it a go.

File: webui/src/lib/components/repository/changes.jsx (hunk 2 above)
Comment (ozkatz) [reply webui/src/lib/components/repository/changes.jsx]: Fixed

File: webui/src/pages/repositories/services/import_data.jsx (hunk 4 above)
Comment (ozkatz) [reply webui/src/pages/repositories/services/import_data.jsx:122]: Nice! changed

File: webui/src/lib/components/repository/changes.jsx (hunk 1 above)
Comment (eladlachmi) [reply webui/src/lib/components/repository/changes.jsx]: > Honestly I just moved existing code and wrapped it in its own component. 

Sorry, did not consider that. Thanks for the fix!

**Review by eladlachmi — APPROVED** [review]
LGTM ✌🏻

**Pull request merged**